patroni/dcs/consul.py

```diff
@@ -370,11 +369,11 @@ def _update_service(self, data):
         role = data['role']
         state = data['state']
         api_parts = urlparse(data['api_url'])
-        api_parts = api_parts._replace(path='/{}'.format(role))
+        api_parts = api_parts._replace(path='/{0}'.format(role))
         conn_parts = urlparse(data['conn_url'])
         check = base.Check.http(api_parts.geturl(), self._service_check_interval, deregister=self._client.http.ttl * 10)
         params = {
-            'service_id': '{}_{}'.format(self._scope, self._name),
+            'service_id': '{0}/{1}'.format(self._scope, self._name),
```

Comment:
I've deliberately changed the delimiter from `'_'` to `'/'`, because usage of underscore may lead to collisions in service_id's across different clusters.